src/tests.rs

```diff
@@ -341,13 +353,19 @@ fn test_drain() {
 
             // range 1..=i
             let (dc, mut sv) = make_segvec_8();
+            to_cleanup.push(dc);
+
             let d = sv.drain((Bound::Included(1), Bound::Included(i)));
             let drained = d.map(|i| i.1).collect::<Vec<_>>();
             assert_eq!(dc.get(), i, "i={}", i);
             assert_eq!(sv.len(), 8 - i, "i={}", i);
             assert_eq!(drained, (2..=(i as i32 + 1)).collect::<Vec<_>>(), "i={}", i);
         }
     }
+
+    for dc in to_cleanup.into_iter() {
```

Comment:
This is fine, but I don't think it is strictly necessary - the memory allocated for these leaked `Box` values is negligible, and this code only runs within the tests, and this cleanup code won't run if the test fails anyway.

Reply:
i agree - it's just that miri will complain about leaked values if these are not deallocated